conu/backend/docker/image.py

```diff
@@ -221,6 +221,51 @@ def push(self, repository=None, tag=None):
                                         self.name, error)
         return image
 
+    def copy(self, repository, tag="latest", source_transport_index=2, target_transport_index=2):
+        """
+        WIP
+        Copy image to repository:tag
+        :param repository to be copied to
+        :param tag
+        :param source_transport_index in transport list WIP. for testing
+                can I find this out tranport from DockerImage instance? maybe useless param
+        :param target_transport_index in tranposrts list WIP, for testing
+        :return: the new DockerImage
+        """
+        transports = ["containers-storage:",
+                      "dir:",
+                      "docker://",
+                      "docker-archive",
+                      "docker-daemon:",
+                      "oci:",
+                      "ostree:"]
+
+        # doing things in local space, maid be deleted later
+        def transport_param(index, repo, tagg):
```

Comment:
I strongly suggest to pull out content of this method into a dedicated class. Nesting functions is really bad, mainly because they can't be tested.

If you don't wish to receive my comments, don't push your changes :D

Reply:
Coments are great, thanks :)
I did not want to keep nested function there forever, but also did not know where to put that function :D

Reply:
create new file ideally, feel free to call it `skopeo.py` or `transport.py` or whatever else fits better